petastorm/reader_impl/worker_loop.py

```diff
@@ -118,10 +160,10 @@ def worker_loop(epochs_generator, loading_pool, loader, decoding_pool, decoder,
             # 3. Read from the shuffling queue and submit decoding tasks to decoding executor
             # ---------------------------------------------------------------------------------
             while shuffling_queue.can_retrieve() and len(in_decoding) < _MAX_IN_DECODING_QUEUE_SIZE:
-                shuffled = shuffling_queue.retrieve()
-                stats['rows_read_from_shuffling_queue'] += 1
-                in_decoding.append(decoding_pool.submit(_decode_row_dispatcher, decoder, shuffled))
-                stats['in_decoding_queue_len'] = len(in_decoding)
+                shuffled = shuffling_queue.retrieve_many(chunk_decoding_size)
+                stats['rows_read_from_shuffling_queue'] += len(shuffled)
+                from_shuffling.update(stats['rows_read_from_shuffling_queue'])
+                in_decoding.append(decoding_pool.submit(_decode_row_dispatcher, decoder, shuffled, row_serializer))
```

Comment:
you are passing `shuffled` (which can be a list) to the `_decode_row_dispatched` method which seems to still expect a single element.

Reply:
It passes the parameters through to the row decoder. Change wrong names in `_decode_row_dispatcher` to reflect that.